src/aiida/orm/nodes/data/structure.py

```diff
@@ -1876,6 +1885,35 @@ def _get_object_pymatgen_molecule(self, **kwargs):
         positions = [list(site.position) for site in self.sites]
         return Molecule(species, positions)
 
+    def to_atomistic(self):
+        """
+        Returns the atomistic StructureData version of the orm.StructureData one.
+        """
+        if not has_atomistic():
+            raise ImportError(
+                'aiida-atomistic plugin is not installed, \
+                please install it to have full support for atomistic structures'
+            )
+        else:
+            from aiida_atomistic import StructureData as AtomisticStructureData
+            from aiida_atomistic import StructureDataMutable as AtomisticStructureDataMutable
+
+        atomistic = AtomisticStructureDataMutable()
+        atomistic.set_pbc(self.pbc)
+        atomistic.set_cell(self.cell)
+
+        for site in self.sites:
+            atomistic.add_atom(
+                **{
+                    'symbols': self.get_kind(site.kind_name).symbol,
+                    'masses': self.get_kind(site.kind_name).mass,
+                    'positions': site.position,
+                    'kinds': site.kind_name,
+                }
+            )
```

Comment:
I'd say it might be more readable if the arguments are directly passed to the function.
```suggestion
                symbols=self.get_kind(site.kind_name).symbol,
                masses= self.get_kind(site.kind_name).mass,
            )
```